// langchain/dist/schema/index.js
import { BaseMessage, HumanMessage, AIMessage, SystemMessage, } from "langchain-core/schema/messages";
import { Serializable } from "../load/serializable.js";
export { RUN_KEY } from "langchain-core/schema/output";
export { BaseMessageChunk, HumanMessageChunk, AIMessageChunk, SystemMessageChunk, FunctionMessage, FunctionMessageChunk, ToolMessage, ToolMessageChunk, ChatMessage, mapStoredMessageToChatMessage, ChatMessageChunk, coerceMessageLikeToMessage, isBaseMessage, isBaseMessageChunk, } from "langchain-core/schema/messages";
export { BaseMessage, HumanMessage, AIMessage, SystemMessage };
/**
 * @deprecated
 * Use {@link BaseMessage} instead.
 */
export const BaseChatMessage = BaseMessage;
/**
 * @deprecated
 * Use {@link HumanMessage} instead.
 */
export const HumanChatMessage = HumanMessage;
/**
 * @deprecated
 * Use {@link AIMessage} instead.
 */
export const AIChatMessage = AIMessage;
/**
 * @deprecated
 * Use {@link SystemMessage} instead.
 */
export const SystemChatMessage = SystemMessage;
export { GenerationChunk, ChatGenerationChunk, } from "langchain-core/schema/output";
export { BasePromptValue } from "langchain-core/schema/prompt";
export { BaseChatMessageHistory, BaseListChatMessageHistory, } from "langchain-core/schema/chat_history";
export { BaseCache } from "langchain-core/schema/cache";
/**
 * Base class for all file stores. All file stores should extend this
 * class.
 */
export class BaseFileStore extends Serializable {
}
/**
 * Base class for all entity stores. All entity stores should extend this
 * class.
 */
export class BaseEntityStore extends Serializable {
}
/**
 * Abstract class for a document store. All document stores should extend
 * this class.
 */
export class Docstore {
}
